cmd/renew_certificate.go

```diff
@@ -168,22 +170,43 @@ func logErrorAndExit(err error) {
 }
 
 func restartControlPlaneService() error {
-	controlPlaneServices := []string{"deploy/dapr-sentry", "deploy/dapr-operator", "statefulsets/dapr-placement-server"}
+	controlPlaneServices := []string{
+		"deploy/dapr-sentry",
+		"deploy/dapr-sidecar-injector",
+		"deploy/dapr-operator",
+		"statefulsets/dapr-placement-server",
+	}
 	namespace, err := kubernetes.GetDaprNamespace()
 	if err != nil {
 		print.FailureStatusEvent(os.Stdout, "Failed to fetch Dapr namespace")
 	}
-	for _, name := range controlPlaneServices {
-		print.InfoStatusEvent(os.Stdout, fmt.Sprintf("Restarting %s..", name))
-		_, err := utils.RunCmdAndWait("kubectl", "rollout", "restart", name, "-n", namespace)
-		if err != nil {
-			return fmt.Errorf("error in restarting deployment %s. Error is %w", name, err)
-		}
-		_, err = utils.RunCmdAndWait("kubectl", "rollout", "status", name, "-n", namespace)
-		if err != nil {
-			return fmt.Errorf("error in checking status for deployment %s. Error is %w", name, err)
-		}
+
+	errs := make([]error, len(controlPlaneServices))
+	var wg sync.WaitGroup
+	wg.Add(len(controlPlaneServices))
+	for i, name := range controlPlaneServices {
+		go func(i int, name string) {
+			defer wg.Done()
+			print.InfoStatusEvent(os.Stdout, fmt.Sprintf("Restarting %s..", name))
+			_, err := utils.RunCmdAndWait("kubectl", "rollout", "restart", "-n", namespace, name)
+			if err != nil {
+				errs[i] = fmt.Errorf("error in restarting deployment %s. Error is %w", name, err)
+				return
+			}
+			_, err = utils.RunCmdAndWait("kubectl", "rollout", "status", "-n", namespace, name)
+			if err != nil {
+				errs[i] = fmt.Errorf("error in checking status for deployment %s. Error is %w", name, err)
+				return
+			}
+		}(i, name)
+	}
+
+	wg.Wait()
+
+	if err := errors.Join(errs...); err != nil {
+		return err
 	}
+
```

Comment:
Sentry and injector have a codependency so need to be rolled out at the same time. Control plane upgrades are considered atomic anyway, so we do all services at the same time.